### PR TITLE
feat: delete old videos Directory

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description

This PR implements functionality to delete outdated video files stored in the previous app directory upon the first launch of the updated app version. This ensures that storage is efficiently managed, preventing unnecessary space usage by obsolete files, and maintaining consistency with the new data storage structure.

fix: LEARNER-9950

### CODE REVIEW

File: dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardViewModel.kt [hunk 1]
```diff
@@ -28,14 +29,16 @@ class DashboardViewModel(
...
+    private val appUpgradeNotifier: AppUpgradeNotifier,
+    private val preferences: CorePreferences
 ) : BaseViewModel() {
 
     private val coursesList = mutableListOf<EnrolledCourse>()
     private var page = 1
     private var isLoading = false
 
     val apiHostUrl get() = config.getApiHostURL()
+    val canResetAppDirectory = preferences.canResetAppDirectory
```
Comment (HamzaIsrar12) [inline dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardViewModel.kt:41]: We will update it as well to `resetAppDirectory`.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Test cases are failing.

File: dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt [hunk 2]
```diff
@@ -161,6 +162,11 @@ class DashboardFragment : Fragment() {
                 )
             }
         }
+
+        if (viewModel.canResetAppDirectory) {
```
[the remark this answers is not in the document]
Comment (omerhabib26) [reply dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt:166]: For now, we can keep it as an independent feature.

File: core/src/main/java/org/openedx/core/utils/FileUtil.kt [hunk 3]
```diff
@@ -13,7 +14,44 @@ object FileUtil {
...
     }
 
+    /**
+     * Deletes all the files and directories in the app's external storage directory.
+     *
+     * @param context The current context.
+     */
+    fun deleteOldAppDirectory(context: Context) {
+        val externalFilesDir = context.getExternalFilesDir(null)
+        val externalAppDir: File = File(externalFilesDir?.parentFile, "videos")
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply core/src/main/java/org/openedx/core/utils/FileUtil.kt]: Please remove warning as well.

**Comment by HamzaIsrar12** [comment]
Test Cases are still failing ...

**Review by HamzaIsrar12 — DISMISSED** [review]

File: app/src/main/java/org/openedx/app/AppActivity.kt [hunk 4]
```diff
@@ -136,6 +136,10 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder {
             }
         }
 
+        if (viewModel.canResetAppDirectory){
+            viewModel.resetAppDirectory(this)
+        }
+
```
Comment (volodymyr-chekyrta) [inline app/src/main/java/org/openedx/app/AppActivity.kt:142]: What if we just encapsulate all this inside of the `AppViewModel#onCreate`?
In this case, we would remove the extra variable and this logic from the view.

Comment (omerhabib26) [reply app/src/main/java/org/openedx/app/AppActivity.kt:142]: sure, I'll update it accordingly.

**Review by k1rill — APPROVED** [review]
LGTM

**Comment by omerhabib26** [comment]
Test cases are failing due to context, the issue is already addressed in [PR-308](https://github.com/openedx/openedx-app-android/pull/308). I'll update and merge it once PR-308 is merged.

**Review by volodymyr-chekyrta — DISMISSED** [review]
LGTM 👍 
Let's merge it after the https://github.com/openedx/openedx-app-android/pull/308

**Pull request closed without merging**